### PR TITLE
update egress gateway instruction

### PR DESCRIPTION
**Pull request by linsun**
to remind users to manually enable egress gateway as it is off by default now.

### CODE REVIEW

**Comment by geeknoid** [comment]
Deploy preview for *preliminary-istio* failed.

Built with commit a0056e1ef47efb2737c314e09464b263070b6273

https://app.netlify.com/sites/preliminary-istio/deploys/5c78a42c054c170009c2d4ab

File: content/boilerplates/before-you-begin-egress-gateway.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (frankbu) [inline content/boilerplates/before-you-begin-egress-gateway.md]: I don't think you need a boilerplate for this.

File: content/docs/examples/advanced-gateways/egress-gateway/index.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ on them cannot access the Internet. Defining an egress gateway, directing all th
 allocating public IPs to the egress gateway nodes allows the application nodes to access external services in a
 controlled way.
 
-{{< boilerplate before-you-begin-egress >}}
+{{< boilerplate before-you-begin-egress-gateway >}}
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress-gateway/index.md]:
```suggestion
{{< boilerplate before-you-begin-egress >}}

{{< warning >}}
If the egress gateway is not enabled, [enable the egress gateway](/docs/examples/advanced-gateways/egress-gateway/index.md#enable-egress-gateway).
{{< /warning >}}
```

File: content/docs/examples/advanced-gateways/egress-gateway/index.md [hunk 3]
```diff
@@ -655,6 +655,18 @@ external service.
 
 1.  Follow the steps in the [Cleanup HTTPS gateway](#cleanup-https-gateway) section.
 
+## Enable Egress Gateway
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress-gateway/index.md]: I don't think this belongs here. It's an awkward fit in the section flow.
How about adding something about how to enable egressgateway in the Install instructions and reference that from the before you begin section?

@vadimeisenbergibm WDYT?

Comment (linsun) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: the question is why would someone wants to read enable egress gateway when they are just doing install...  I'm open to ideas on where it should belong to :)

File: content/docs/examples/advanced-gateways/egress-gateway/index.md (hunk 2 above)
Comment (linsun) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: are you okay to have this warning for other egress tasks, which you don't necessarily need the egress gateway?

for example, I didn't want to give out the warning for this task as it is not needed:
for example:  https://preliminary.istio.io/docs/tasks/traffic-management/egress/

Comment (frankbu) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: Right, that's why I put the warning after the boilerplate, not inside it.

File: content/docs/examples/advanced-gateways/egress-gateway/index.md (hunk 3 above)
Comment (frankbu) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: I know the install instructions need and are in the process of getting a lot of changes, but one thing I think is missing in the docs is what the various profiles (minimal, demo, standard) includes by default. For example the minimal install doc needs to say more than `This minimal install provides traffic management features of Istio`. What exactly is/isn't included. The "traffic management features" sort of implies the egressgateway would be included. I see that in the [helm option doc](https://preliminary.istio.io/docs/reference/config/installation-options/), it says egressgateway is enabled by default, so we probably need to clarify what "default" means.

Some of the install instructions are also missing a "Verify Installation" step that does the `kubectl get po` to confirm all the expected components are up and running. Anyway, I think our install instructions need a lot of work including somewhere reasonable to reference from examples that tell people about how to enable/disable the standard addons, like egressgateway.

File: content/docs/examples/advanced-gateways/egress-gateway/index.md (hunk 2 above)
Comment (linsun) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: ah good point!  thanks!!

File: content/docs/examples/advanced-gateways/egress-gateway/index.md (hunk 3 above)
Comment (linsun) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: so the helm option doc will need to be refreshed... (cc @johnma14) 

egress gateway is off by default, unless you use demo profile. That is a change went in a week ago.

Comment (linsun) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: I agree, if there is a page that talks about these are default components, and if you want to enable these components, here is what you do.  The current disabled (or to be disabled) components in v1.1 are istio-pilocy and istio-egressgateway and a few addons.

Comment (rcaballeromx) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: @linsun The recent change to the install docs made several issues very clear. We definitely need to add the guidance that you and @frankbu suggest, but it is important that we do so in a consistent and systematic way to ensure we provide correct information without adding too much maintenance overhead. 
Feel free to check out some of the bugs I've already filed and comment.
https://github.com/istio/istio.io/issues/3431
https://github.com/istio/istio.io/issues/3432
https://github.com/istio/istio.io/issues/3433

@frankbu discussed it and he is adding an issue specifically for this concern.

Comment (frankbu) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: #3460

Comment (vadimeisenbergibm) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: @linsun @frankbu The issue is simple, you do not need to instal/update the whole Istio. You just deploy the egress gateway alone, by the following command:

```
$ helm template $HOME/istio-fetch/istio --name istio-egressgateway --namespace istio-system -x charts/gateways/templates/deployment.yaml -x charts/gateways/templates/service.yaml -x charts/gateways/templates/serviceaccount.yaml -x charts/gateways/templates/autoscale.yaml -x charts/gateways/templates/clusterrole.yaml -x charts/gateways/templates/clusterrolebindings.yaml --set global.istioNamespace=istio-system --set gateways.istio-ingressgateway.enabled=false --set gateways.istio-egressgateway.enabled=true | kubectl apply -f -
serviceaccount "istio-egressgateway-service-account" created
clusterrole "istio-egressgateway-istio-system" created
clusterrolebinding "istio-egressgateway-istio-system" created
service "istio-egressgateway" created
deployment "istio-egressgateway" created
horizontalpodautoscaler "istio-egressgateway" created
```

Comment (frankbu) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: I think we can do something better in the future, but maybe a good solution for now is to just add the command to the {{< warning >>}}, above.
```
{{< warning >}}
If the egress gateway is not enabled, run the following command to enable it:
{{< text bash >}}
$ helm template $HOME/istio-fetch/istio --name istio-egressgateway --namespace istio-system -x charts/gateways/templates/deployment.yaml -x charts/gateways/templates/service.yaml -x charts/gateways/templates/serviceaccount.yaml -x charts/gateways/templates/autoscale.yaml -x charts/gateways/templates/clusterrole.yaml -x charts/gateways/templates/clusterrolebindings.yaml --set global.istioNamespace=istio-system --set gateways.istio-ingressgateway.enabled=false --set gateways.istio-egressgateway.enabled=true | kubectl apply -f -
{{< /text >}}
{{< /warning >}}
```

Comment (linsun) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: @frankbu I think that is probably good for now I will make the editing.

Comment (linsun) [reply content/docs/examples/advanced-gateways/egress-gateway/index.md]: validated the cmd runs good on my env. :)

File: content/docs/examples/advanced-gateways/egress-gateway/index.md [hunk 4]
```diff
@@ -31,6 +31,15 @@ controlled way.
 
 {{< boilerplate before-you-begin-egress >}}
 
+{{< warning >}}
+{{< warning >}}
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress-gateway/index.md]: delete line

File: content/docs/examples/advanced-gateways/egress-gateway/index.md [hunk 5]
```diff
@@ -31,6 +31,15 @@ controlled way.
...
+If the egress gateway is not enabled, run the following command to enable it:
+{{< text bash >}}
+$ helm template $HOME/istio-fetch/istio --name istio-egressgateway --namespace istio-system -x charts/gateways/templates/deployment.yaml -x charts/gateways/templates/service.yaml -x charts/gateways/templates/serviceaccount.yaml -x charts/gateways/templates/autoscale.yaml -x charts/gateways/templates/clusterrole.yaml -x charts/gateways/templates/clusterrolebindings.yaml --set global.istioNamespace=istio-system --set gateways.istio-ingressgateway.enabled=false --set gateways.istio-egressgateway.enabled=true | kubectl apply -f -
+{{< /text >}}
+{{< /warning >}}
+{{< /warning >}}
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress-gateway/index.md]: delete line

File: content/docs/examples/advanced-gateways/egress-gateway/index.md [hunk 6]
```diff
@@ -31,6 +31,15 @@ controlled way.
 
 {{< boilerplate before-you-begin-egress >}}
 
+{{< warning >}}
+{{< warning >}}
+If the egress gateway is not enabled, run the following command to enable it:
+{{< text bash >}}
+$ helm template $HOME/istio-fetch/istio --name istio-egressgateway --namespace istio-system -x charts/gateways/templates/deployment.yaml -x charts/gateways/templates/service.yaml -x charts/gateways/templates/serviceaccount.yaml -x charts/gateways/templates/autoscale.yaml -x charts/gateways/templates/clusterrole.yaml -x charts/gateways/templates/clusterrolebindings.yaml --set global.istioNamespace=istio-system --set gateways.istio-ingressgateway.enabled=false --set gateways.istio-egressgateway.enabled=true | kubectl apply -f -
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress-gateway/index.md]:
```suggestion
$ helm template $HOME/istio-fetch/istio --name istio-egressgateway --namespace istio-system \
    -x charts/gateways/templates/deployment.yaml -x charts/gateways/templates/service.yaml \
    -x charts/gateways/templates/serviceaccount.yaml -x charts/gateways/templates/autoscale.yaml \
    -x charts/gateways/templates/clusterrole.yaml -x charts/gateways/templates/clusterrolebindings.yaml \
    --set global.istioNamespace=istio-system --set gateways.istio-ingressgateway.enabled=false \
    --set gateways.istio-egressgateway.enabled=true | kubectl apply -f -
```

**Comment by linsun** [comment]
@frankbu could you approve?  The error in circle isn't related to this PR.

**Comment by linsun** [comment]
I don't mind adding it (I had it earlier) but there was concerns on where to add it from @frankbu earlier on.

**Comment by linsun** [comment]
can we get this merged first?  I think having a warning before start makes sense.

**Comment by vadimeisenbergibm** [comment]
@linsun OK, let's merge it, I will submit another PR to make it a separate section and reference it in other examples.

**Pull request merged**